File: leetcode-venture/2191.py
class Solution:
    def sortJumbled(self, mapping: List[int], nums: List[int]) -> List[int]:
        n2j = {}
        counts = defaultdict(int)
        res = []

        mapping = [str(m) for m in mapping]
        nums = [str(num) for num in nums]
        
        for x in nums:
            counts[int(x)] += 1
            cand = ""
            for c in x:
                cand += mapping[int(c)]
            n2j[int(x)] = int(cand)
        
        tmp = list(n2j.items())
        tmp.sort(key=lambda x:x[1])

        for x in tmp:
            while counts[x[0]]:
                res.append(x[0])
                counts[x[0]] -= 1
        
        return res

File: leetcode/2191.cpp
class Solution
{
public:
    vector<int> sortJumbled(vector<int> &mapping, vector<int> &nums)
    {
        int n = nums.size();
        vector<pair<int, int>> arr(n);
        for (int i = 0; i < n; ++i)
        {
            string s = to_string(nums[i]);
            for (char &c : s)
            {
                c = mapping[c - '0'] + '0';
            }
            arr[i] = {stoi(s), i};
        }
        sort(arr.begin(), arr.end());
        vector<int> ret(n);
        for (int i = 0; i < n; ++i)
        {
            ret[i] = nums[arr[i].second];
        }
        return ret;
    }
};
